### PR TITLE
Test navigation actions

### PR DESCRIPTION
**Pull request by ThomasMPont**
## Changes

`Screen.kt` now contains two extra dummy routes for testing purposes.

`NavigationActions.navigate(Route)` has been reverted to redirect to the `UnauthenticatedScreen()` when the user is not authorized. This avoids using `Context` in `NavigationActions`.

All navigation logic related files have been moved from the `ui/` package to `model/` for consistency.

## Tests

All methods from `NavigationActions.kt` now have unit tests. Previously, none were tested.

## Future additions

The `UnauthenticatedScreen()` still looks a little clunky. Maybe we could make it a popup or decorate it a little better in the future.

### CODE REVIEW

File: app/src/test/java/com/github/se/signify/model/navigation/NavigationActionsTest.kt
Comment (GRGool) [inline app/src/test/java/com/github/se/signify/model/navigation/NavigationActionsTest.kt:29]: I dont think this naming is intended right? Its not aligned with the rest.

**Review by GRGool — CHANGES_REQUESTED** [review]
The PR is straight forward, provides testing and basic coverage for the navigation logic of auth/unauth users, good job. Small issue with the naming of one of the tests that doesn't align with the rest.

File: app/src/test/java/com/github/se/signify/model/navigation/NavigationActionsTest.kt
Comment (ThomasMPont) [reply app/src/test/java/com/github/se/signify/model/navigation/NavigationActionsTest.kt]: Good catch. I forgot to delete this temporary test of a test.

**Comment by ThomasMPont** [comment]
@GRGool 
I've addressed your request for changes, and have added a couple new tests. Could you take a look at my latest commits?

**Comment by GRGool** [comment]
All good for me, good additions for the restricted TopLevelDest tests, as well as the currentRoute test(I checked navigationsActions to see if the functions were all tested but just skipped this one mb)

**Comment by GRGool** [comment]
Uh theres conflicts you should resolve them right?

**Review by GRGool — APPROVED** [review]

**Comment by GRGool** [comment]
Was waiting for the CI to run, approved you can merge it when the conflicts are resolved

**Pull request merged**
